src/main/plug/sampler_kernel.cpp

```diff
@@ -670,6 +685,11 @@ namespace lsp
             afsample_t *afs     = af->vData[AFI_CURR];
             if (afs->pSource != NULL)
             {
+                afs->pSample->copy(afs->pSource);
+                nSampleRate = af->fSampleRate * 1000;
+                afs->pSource->resample(af->fSampleRate * 1000);
```

Comment:
Trying to modify source sample is a bad idea. Instead of this, I would like to suggest to:
- remove the resampling from the `sampler_kernel::load_file` method;
- create temporary sample object and copy data from the `afs->pSample`;
- resample the copied sample to the proper sample rate computed here.